instana/collector/aws_eks_fargate.py

```diff
@@ -0,0 +1,53 @@
+# (c) Copyright IBM Corp. 2023
+
+"""
+Collector for EKS Pods on AWS Fargate: Manages the periodic collection of metrics & snapshot data
+"""
+
+from time import time
+from instana.log import logger
+from instana.collector.base import BaseCollector
+from instana.collector.helpers.eks.process import EKSFargateProcessHelper
+from instana.collector.helpers.runtime import RuntimeHelper
+from instana.util import DictionaryOfStan
+
+
+class EKSFargateCollector(BaseCollector):
+    """ Collector for EKS Pods on AWS Fargate """
+
+    def __init__(self, agent):
+        super(EKSFargateCollector, self).__init__(agent)
+        logger.debug("Loading Collector for EKS Pods on AWS Fargate ")
+
+        self.snapshot_data = DictionaryOfStan()
+        self.snapshot_data_sent = False
+        self.podname = agent.podname
+        self.helpers.append(EKSFargateProcessHelper(self))
+        self.helpers.append(RuntimeHelper(self))
+
+    def should_send_snapshot_data(self):
+        return int(time()) - self.snapshot_data_last_sent > self.snapshot_data_interval
+
+    def prepare_payload(self):
+        payload = DictionaryOfStan()
+        payload["spans"] = []
+        payload["metrics"]["plugins"] = []
+
+        try:
+            if not self.span_queue.empty():
+                payload["spans"] = self.queued_spans()
+
+            with_snapshot = self.should_send_snapshot_data()
+
+            plugins = []
+            for helper in self.helpers:
+                plugins.extend(helper.collect_metrics(with_snapshot=with_snapshot))
+
+            payload["metrics"]["plugins"] = plugins
+
+            if with_snapshot is True:
```

Comment:
Shouldn't we just say `if with_snapshot:`  since `should_send_snapshot_data()` always returns a bool?

Reply:
We should. Good catch!